lib/shared/get-cache-filename.js

```diff
@@ -1,9 +1,53 @@
+const fs = require('fs');
 const path = require('path');
 
-module.exports = function getCacheFilename(modelConfig) {
+function getCacheFilename(modelConfig, newFile = false) {
     let { slug, cache } = modelConfig;
+    let cacheFiles = [];
+    let cacheFile;
+
+    if (fs.existsSync(cache) && !newFile) {
+        cacheFiles = getCacheFiles(cache)[slug] || [];
+
+        if (cacheFiles.length) {
+            cacheFile = cacheFiles[0].file;
+        }
+    }
 
     return slug && cache
-        ? path.join(cache, `.discoveryjs.${slug}.cache`)
+        ? path.join(cache, cacheFile || `.discoveryjs.${slug}.${Date.now()}.cache`)
         : undefined;
 };
+
+function getCacheFiles(cacheDir) {
```

Comment:
What if to add a model list (names) here. Otherwise the server (for instance) will process caches for models that out of scope, i.e. not in config or launched with `--model`